pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -60,25 +63,41 @@ var (
 	errInconsistentPodSetAssignments = errors.New("inconsistent podSet assignments")
 )
 
-var (
-	MaxRetries        int32 = 3
-	MinBackoffSeconds int32 = 60
-)
-
 type provisioningConfigHelper = admissioncheck.ConfigHelper[*kueue.ProvisioningRequestConfig, kueue.ProvisioningRequestConfig]
 
 func newProvisioningConfigHelper(c client.Client) (*provisioningConfigHelper, error) {
 	return admissioncheck.NewConfigHelper[*kueue.ProvisioningRequestConfig](c)
 }
 
+type waitForProvisioningRequestReadyConfig struct {
+	maxRetries        int32
+	minBackoffSeconds int32
+	maxBackoffSeconds int32
+}
+
 type Controller struct {
-	client client.Client
-	helper *provisioningConfigHelper
-	record record.EventRecorder
+	client                          client.Client
+	record                          record.EventRecorder
+	helper                          *provisioningConfigHelper
+	waitForProvisioningRequestReady *waitForProvisioningRequestReadyConfig
 }
 
 var _ reconcile.Reconciler = (*Controller)(nil)
 
+// Option configures the provisioning controller.
+type Option func(*Controller)
+
+func WithWaitForProvisioningMaxRetries(maxRetries int32) Option {
+	return func(controller *Controller) {
+		controller.waitForProvisioningRequestReady.maxRetries = maxRetries
+	}
+}
```

Comment:
```suggestion
type Option func(*options)

type options struct {
	waitForProvisioningRequestReady *waitForProvisioningRequestReady
}

func WithWaitForProvisioningMaxRetries(maxRetries int32) Option {
	return func(o *options) {
		o.waitForProvisioningRequestReady = maxRetries
	}
}
```

Could you avoid exposing unnecessary fields in the Controller?
Additionally, we can avoid null pointers by such dedicated typed `options`.

Reply:
Done